Docs(setup) Big re-org of Setup nav, along with minor re-writes

dorbin opened the pull request.

dorbin committed:
docs(Setup and Concepts) de-dupe Providers and Storage overviews and add Providers concept

dorbin committed:
docs(Setup) Big re-org of Setup nav, plus some minor re-writes

dorbin committed:
docs(Setup) Big re-org of Setup nav, plus some minor re-writes

skim1420 commented:
Sorry for the delay, let's please review together. Hard to get a feel for how well the nav structure works looking at underlying files.

Also, if you could add some set of screenshot, description/jist, rationale summary, etc. that would help too.

dorbin commented:
@skim1420 ,

The main change here is to the nav for /setup/. Currently it's not procedural at the high level, and it references sections (like bakery and CI) that are lower down on the nav. In fact, the nav is too long, for this reason.

In an earlier PR I moved Storage and Providers up into the Install nav and removed the related sections that were lower down. This PR does something similar to bakery, ci, monitoring, features, and it makes the main install/config procedural. That is, the parts that are stepwise are now numbered. 

After this PR, it'll be divided up into quickstarts, install, and "configure everything else." Additionally, it takes advantage of change Scott made to allow collapsable nav headings. 

Screenshots to come...

dorbin commented:
![setup_nav_collapsed](https://user-images.githubusercontent.com/29312809/34846586-9b81e310-f6e6-11e7-9dc0-f299e55bc05e.png)
![setup_nav_everything_else](https://user-images.githubusercontent.com/29312809/34846587-9b89b112-f6e6-11e7-9295-c94c6bdcfb5b.png)
![setup_nav_install](https://user-images.githubusercontent.com/29312809/34846588-9b90b796-f6e6-11e7-946c-baeb1359ed3f.png)

dorbin committed:
docs(Community) Refinements to GitHub entry on main Community page (#477)

dorbin committed:
Merge branch 'master' of https://github.com/dorbin/spinnaker.github.io

dorbin committed:
Merge remote-tracking branch 'upstream/master'
